contrib/platform/test/com/sun/jna/platform/win32/Kernel32Test.java

```diff
@@ -372,14 +373,14 @@ public void testGetCurrentThreadId() {
     public void testGetCurrentThread() {
         HANDLE h = Kernel32.INSTANCE.GetCurrentThread();
         assertNotNull("No current thread handle", h);
-        assertFalse("Null current thread handle", h.equals(0));
+        assertFalse("Null current thread handle", h.getPointer().equals(Pointer.NULL));
```

Comment:
Please make this 
```java
assertNotNull("Null current thread handle", h.getValue());
```

Your version will raise a `NullPointerException` in the failure case. I consider `Pointer.NULL` a very bad idea - it gives the illusion as if it were an object, in fact it is `null`. If you want to keep your version, it needs to be:

```java
assertFalse("Null current thread handle", h.getPointer() == Pointer.NULL);
```

Reply:
The irony is that the first time I typed that solution I thought "This won't work in this order as it gives an NPE" and backspaced and dug through the code to try to find an alternative, and ended up in the same place!  Oops.

Reply:
It doesn't seem like `h.getValue()` is available.  Using `h.getPointer()`.